### ISSUE TITLE
[FEATURE] Ability to work on Wikibase Client

### ISSUE THREAD

**Issue by lucamauri** [ProfessionalWiki/SemanticWikibase#2]
If I understand the instructions properly, this extension will only work if the Semantic MediaWiki (and the extension) are installed on a MediaWiki instance with WikiBase Repository.
If I am correct, what about allowing this extension to work on a MW isntance with WikiBase Client on IT?

I think it is a common setup to have two instances of MediaWiki: one that only act as Repository and one that acts as Client, this basically replicates the behavior of WIkiData and the different languages' Wikipedia.
In my case, for instance, I have a MW installation that only act as Repository and a MW instance where Client is installed: SMW in my case is installed on the Client instance and it's there that I would also be able to use SemanticWikibase

Thanks

Luca

**Comment by JeroenDeDauw** [comment]
Correct, this extension needs Wikibase Repository and has no integration with Wikibase Client.

Integration with Wikibase Client is more complicated. We'd need to listen to changes made by the connected repository to keep the SMW copy of the data up to date. And we'd need to deal with SMW data rebuilds, which otherwise would make the data go poof, since it is not in the wiki itself.

If there is demand and funding for Wikibase Client integration, I'd be interested in working on this. I'm also happy to provide some assistance for other developers to create it, but beware, this is not a trivial task, especially if you are not familiar with SMW.
